File: app/src/main/java/de/sneakpeek/data/MovieInfoCrew.java
package de.sneakpeek.data;


import android.os.Parcel;
import android.os.Parcelable;

import com.squareup.moshi.Json;

public class MovieInfoCrew implements Parcelable {

    public final static Parcelable.Creator<MovieInfoCrew> CREATOR = new Creator<MovieInfoCrew>() {

        public MovieInfoCrew createFromParcel(Parcel in) {
            MovieInfoCrew instance = new MovieInfoCrew();
            instance.name = ((String) in.readValue((String.class.getClassLoader())));
            instance.job = ((String) in.readValue((String.class.getClassLoader())));
            return instance;
        }

        public MovieInfoCrew[] newArray(int size) {
            return (new MovieInfoCrew[size]);
        }

    };

    @Json(name = "name") public String name;
    @Json(name = "job") public String job;

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(name);
        dest.writeValue(job);
    }

    public int describeContents() {
        return 0;
    }
}

File: app/src/main/java/de/sneakpeek/data/MovieInfoCrew.kt
package de.sneakpeek.data


import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class MovieInfoCrew(
        @Expose @SerializedName("name") var name: String?,
        @Expose @SerializedName("job") var job: String?) : Parcelable {

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<MovieInfoCrew> = object : Parcelable.Creator<MovieInfoCrew> {
            override fun createFromParcel(source: Parcel): MovieInfoCrew = MovieInfoCrew(source)
            override fun newArray(size: Int): Array<MovieInfoCrew?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this(
            source.readString(),
            source.readString()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeString(job)
    }
}


